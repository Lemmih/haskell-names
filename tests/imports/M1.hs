module M1 where

import Prelude
