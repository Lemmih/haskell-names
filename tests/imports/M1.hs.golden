Table (
  fromList
    [ ( GName
          ""
          "Constructor1"
      , fromList
          [ SymConstructor
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "Constructor1"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataType"
                                }
              }
          ]
      )
    , ( GName
          ""
          "Constructor2"
      , fromList
          [ SymConstructor
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "Constructor2"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataType"
                                }
              }
          ]
      )
    , ( GName
          ""
          "DataTypeWithSelectors"
      , fromList
          [ SymConstructor
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataTypeWithSelectors"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataTypeWithSelectors"
                                }
              }
          ]
      )
    , ( GName
          ""
          "Newtype"
      , fromList
          [ SymConstructor
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "Newtype"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "Newtype"
                                }
              }
          ]
      )
    , ( GName
          ""
          "NewtypeWithSelectors"
      , fromList
          [ SymConstructor
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "NewtypeWithSelectors"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "NewtypeWithSelectors"
                                }
              }
          ]
      )
    , ( GName
          ""
          "function"
      , fromList
          [ SymValue
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "function"
                                }
              , sv_fixity = Nothing
              }
          ]
      )
    , ( GName
          ""
          "method1"
      , fromList
          [ SymMethod
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "method1"
                                }
              , sv_fixity = Nothing
              , sv_className = OrigName
                                 { origPackage = Nothing
                                 , origGName = GName
                                                 "Library"
                                                 "Class"
                                 }
              }
          ]
      )
    , ( GName
          ""
          "method2"
      , fromList
          [ SymMethod
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "method2"
                                }
              , sv_fixity = Nothing
              , sv_className = OrigName
                                 { origPackage = Nothing
                                 , origGName = GName
                                                 "Library"
                                                 "Class"
                                 }
              }
          ]
      )
    , ( GName
          ""
          "selector1"
      , fromList
          [ SymSelector
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "selector1"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataTypeWithSelectors"
                                }
              }
          ]
      )
    , ( GName
          ""
          "selector2"
      , fromList
          [ SymSelector
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "selector2"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataTypeWithSelectors"
                                }
              }
          ]
      )
    , ( GName
          ""
          "unNewtype"
      , fromList
          [ SymSelector
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "unNewtype"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "NewtypeWithSelectors"
                                }
              }
          ]
      )
    , ( GName
          "Library"
          "Constructor1"
      , fromList
          [ SymConstructor
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "Constructor1"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataType"
                                }
              }
          ]
      )
    , ( GName
          "Library"
          "Constructor2"
      , fromList
          [ SymConstructor
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "Constructor2"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataType"
                                }
              }
          ]
      )
    , ( GName
          "Library"
          "DataTypeWithSelectors"
      , fromList
          [ SymConstructor
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataTypeWithSelectors"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataTypeWithSelectors"
                                }
              }
          ]
      )
    , ( GName
          "Library"
          "Newtype"
      , fromList
          [ SymConstructor
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "Newtype"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "Newtype"
                                }
              }
          ]
      )
    , ( GName
          "Library"
          "NewtypeWithSelectors"
      , fromList
          [ SymConstructor
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "NewtypeWithSelectors"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "NewtypeWithSelectors"
                                }
              }
          ]
      )
    , ( GName
          "Library"
          "function"
      , fromList
          [ SymValue
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "function"
                                }
              , sv_fixity = Nothing
              }
          ]
      )
    , ( GName
          "Library"
          "method1"
      , fromList
          [ SymMethod
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "method1"
                                }
              , sv_fixity = Nothing
              , sv_className = OrigName
                                 { origPackage = Nothing
                                 , origGName = GName
                                                 "Library"
                                                 "Class"
                                 }
              }
          ]
      )
    , ( GName
          "Library"
          "method2"
      , fromList
          [ SymMethod
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "method2"
                                }
              , sv_fixity = Nothing
              , sv_className = OrigName
                                 { origPackage = Nothing
                                 , origGName = GName
                                                 "Library"
                                                 "Class"
                                 }
              }
          ]
      )
    , ( GName
          "Library"
          "selector1"
      , fromList
          [ SymSelector
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "selector1"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataTypeWithSelectors"
                                }
              }
          ]
      )
    , ( GName
          "Library"
          "selector2"
      , fromList
          [ SymSelector
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "selector2"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataTypeWithSelectors"
                                }
              }
          ]
      )
    , ( GName
          "Library"
          "unNewtype"
      , fromList
          [ SymSelector
              { sv_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "unNewtype"
                                }
              , sv_fixity = Nothing
              , sv_typeName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "NewtypeWithSelectors"
                                }
              }
          ]
      )
    ] ) (
  fromList
    [ ( GName
          ""
          "Class"
      , fromList
          [ SymClass
              { st_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "Class"
                                }
              , st_fixity = Nothing
              }
          ]
      )
    , ( GName
          ""
          "DataType"
      , fromList
          [ SymData
              { st_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataType"
                                }
              , st_fixity = Nothing
              }
          ]
      )
    , ( GName
          ""
          "DataTypeWithSelectors"
      , fromList
          [ SymData
              { st_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataTypeWithSelectors"
                                }
              , st_fixity = Nothing
              }
          ]
      )
    , ( GName
          ""
          "Newtype"
      , fromList
          [ SymNewType
              { st_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "Newtype"
                                }
              , st_fixity = Nothing
              }
          ]
      )
    , ( GName
          ""
          "NewtypeWithSelectors"
      , fromList
          [ SymNewType
              { st_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "NewtypeWithSelectors"
                                }
              , st_fixity = Nothing
              }
          ]
      )
    , ( GName
          ""
          "TypeSynonym"
      , fromList
          [ SymType
              { st_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "TypeSynonym"
                                }
              , st_fixity = Nothing
              }
          ]
      )
    , ( GName
          "Library"
          "Class"
      , fromList
          [ SymClass
              { st_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "Class"
                                }
              , st_fixity = Nothing
              }
          ]
      )
    , ( GName
          "Library"
          "DataType"
      , fromList
          [ SymData
              { st_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataType"
                                }
              , st_fixity = Nothing
              }
          ]
      )
    , ( GName
          "Library"
          "DataTypeWithSelectors"
      , fromList
          [ SymData
              { st_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "DataTypeWithSelectors"
                                }
              , st_fixity = Nothing
              }
          ]
      )
    , ( GName
          "Library"
          "Newtype"
      , fromList
          [ SymNewType
              { st_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "Newtype"
                                }
              , st_fixity = Nothing
              }
          ]
      )
    , ( GName
          "Library"
          "NewtypeWithSelectors"
      , fromList
          [ SymNewType
              { st_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "NewtypeWithSelectors"
                                }
              , st_fixity = Nothing
              }
          ]
      )
    , ( GName
          "Library"
          "TypeSynonym"
      , fromList
          [ SymType
              { st_origName = OrigName
                                { origPackage = Nothing
                                , origGName = GName
                                                "Library"
                                                "TypeSynonym"
                                }
              , st_fixity = Nothing
              }
          ]
      )
    ] )